wallet/src/Dscp/Wallet/CLI.hs

```diff
@@ -1,31 +1,46 @@
 module Dscp.Wallet.CLI
-       ( WalletCLIParams (..)
-       , getWalletCLIParams
+       ( getWalletConfig
+       , getCLIWalletConfig
        ) where
 
 import Options.Applicative (execParser, fullDesc, help, helper, info, long, metavar, optional,
-                            progDesc, strOption)
+                            progDesc, strOption, Parser)
 
+import Loot.Config (OptModParser, (.::), (.:<))
 import Dscp.CommonCLI
 import Dscp.Config
-import Dscp.Web
+import Dscp.Wallet.Config
 
-data WalletCLIParams = WalletCLIParams
-    { wpWitness      :: BaseUrl
-    , wpKnitCommand  :: Maybe Text
-    , wpConfigParams :: ConfigParams
-    }
+knitCommandParser :: Parser (Maybe Text)
+knitCommandParser = optional $ strOption $
+    long "knit" <>
+    metavar "COMMAND" <>
+    help "Execute provided knit command and exit."
 
-getWalletCLIParams :: IO WalletCLIParams
-getWalletCLIParams = do
-    let parser = do
-            wpWitness <- clientAddressParser "witness" "Address of a witness node to communicate with"
-            wpKnitCommand <- optional $ strOption $
-                long "knit" <>
-                metavar "COMMAND" <>
-                help "Execute provided knit command and exit."
-            wpConfigParams <- configParamsParser
-            return WalletCLIParams{..}
-    execParser $
+walletConfigParser :: OptModParser WalletConfig
+walletConfigParser = #wallet .:<
+    (#witness .:: clientAddressParser "witness"
+                  "Address of a witness node to communicate with")
+
+getWalletConfig :: IO WalletConfigRec
+getWalletConfig = do
+    let parser = (,) <$> configParamsParser <*> walletConfigParser
+    (configParams, cliConfigMod) <- execParser $
         info (helper <*> versionOption <*> parser) $
         fullDesc <> progDesc "Ariadne wallet for Disciplina"
+    let wrapConfig cfg = cliConfigMod $ defaultWalletConfig <> cfg
+    buildConfig configParams $
+        fmap wrapConfig . fillWalletConfig
+
+getCLIWalletConfig :: IO (WalletConfigRec, Maybe Text)
+getCLIWalletConfig = do
+    let parser = (,,) <$> configParamsParser <*>
+                          walletConfigParser <*>
+                          knitCommandParser
+    (configParams, cliConfigMod, knitCommand) <- execParser $
+        info (helper <*> versionOption <*> parser) $
+        fullDesc <> progDesc "Ariadne cli wallet for Disciplina"
+    let wrapConfig cfg = cliConfigMod $ defaultWalletConfig <> cfg
+    configRec <- buildConfig configParams $
+        fmap wrapConfig . fillWalletConfig
```

Comment:
This copy-paste between `getWalletConfig` and `getCLIWalletConfig` can be for sure reduced.